clippy_dev/src/main.rs

```diff
@@ -87,6 +87,28 @@ fn print_lints() {
 
 fn update_lints(update_mode: &UpdateMode) {
     let lint_list: Vec<Lint> = gather_all().collect();
+
+    std::fs::write(
+        "../src/lintlist.rs",
+        &format!(
+            "\
+/// Lint data parsed from the Clippy source code.
+#[derive(Clone, PartialEq, Debug)]
+pub struct Lint {{
+    pub name: &'static str,
+    pub group: &'static str,
+    pub desc: &'static str,
+    pub deprecation: Option<&'static str>,
+    pub module: &'static str,
+}}
+
+pub const ALL_LINTS: [Lint; {}] = {:#?};",
+            lint_list.len(),
```

Comment:
I believe this should use `usable_lints` from below, otherwise it will include internal and deprecated lints.

Reply:
would we never want to list deprecated lints in the -W help text?

Reply:
Never, yes.

If someone does `-W deprecatedlintname` it would be nice to say it's renamed, but this isn't necessary